scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala

```diff
@@ -298,7 +303,62 @@ object node {
   // shape of this data should probably change, currently we leave it for backward compatibility
   object FragmentInputDefinition {
 
-    @JsonCodec case class FragmentParameter(name: String, typ: FragmentClazzRef)
+    @JsonCodec case class FixedExpressionValue(expression: String, label: String)
+
+    object FragmentParameterInputMode extends Enumeration {
+      type FragmentParameterInputMode = Value
+
+      implicit val typeEncoder: Encoder[FragmentParameterInputMode.Value] =
+        Encoder.encodeEnumeration(FragmentParameterInputMode)
+      implicit val typeDecoder: Decoder[FragmentParameterInputMode.Value] =
+        Decoder.decodeEnumeration(FragmentParameterInputMode)
+
+      val InputModeAny: Value                = Value("InputModeAny")
+      val InputModeAnyWithSuggestions: Value = Value("InputModeAnyWithSuggestions")
+      val InputModeFixedList: Value          = Value("InputModeFixedList")
+    }
+
+    object FragmentParameter {
+
+      def apply(name: String, typ: FragmentClazzRef): FragmentParameter = {
+        FragmentParameter(
+          name,
+          typ,
+          required = false,
+          initialValue = None,
+          hintText = None,
+          inputConfig = FragmentParameterInputConfig(InputModeAny, None)
+        )
+      }
+
+    }
+
+    @JsonCodec
+    case class FragmentParameter(
+        name: String,
+        typ: FragmentClazzRef,
+        required: Boolean,
```

Comment:
This is persisted in the db as part of a fragment json. Now that this class expects more fields without default or optional values like `inputConfig` and `required` when running the application with database with persisted fragment in old format without these fields, there is a decoding error and the application crashes. In the current form this would require applying a migration.
```
java.lang.IllegalArgumentException: Unmarshalling errors: DecodingFailure at .nodes[0].parameters.required: Missing required field
	at pl.touk.nussknacker.engine.marshall.ProcessMarshaller$.$anonfun$fromJsonUnsafe$1(ProcessMarshaller.scala:121)
```